File: app/src/main/java/cf/movie/slmovie/base/BaseMovies/model/BaseMoviesAdapter.java
package cf.movie.slmovie.base.BaseMovies.model;

import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;

import java.util.ArrayList;

import cf.movie.slmovie.R;
import cf.movie.slmovie.base.BaseMovies.bean.BaseMoviesBean;

/**
 * Created by 包俊 on 2017/7/21.
 */

public class BaseMoviesAdapter extends RecyclerView.Adapter<BaseMoviesAdapter.HotMovieViewHolder> {

    private ArrayList<BaseMoviesBean.movies> movies;

    public BaseMoviesAdapter(ArrayList<BaseMoviesBean.movies> movies) {
        this.movies = movies;
    }

    public void refresh(ArrayList<BaseMoviesBean.movies> movies) {
        this.movies = movies;
        notifyDataSetChanged();
    }

    @Override
    public HotMovieViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_hotmovie, parent, false);
        return new HotMovieViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(HotMovieViewHolder holder, int position) {
        BaseMoviesBean.movies movie = movies.get(position);
        Uri uri = Uri.parse(movie.getPost());
        holder.simpleDraweeView.setImageURI(uri);
        holder.tv_name.setText(movie.getName());
        holder.tv_year.setText("上映年代：" + movie.getYear());
        holder.tv_location.setText("地区：" + movie.getDetails().getLocation());
        holder.tv_douban.setText("评分：" + movie.getDouban());
        holder.tv_actor.setText("主演：" + movie.getDetails().getActor());
        if (movie.getDetails().isTV()) {
            holder.tv_status.setText(movie.getDetails().getStatus());
            holder.tv_type.setVisibility(View.GONE);
            holder.tv_status.setVisibility(View.VISIBLE);
        } else {
            holder.tv_type.setText("类型：" + movie.getDetails().getType());
            holder.tv_type.setVisibility(View.VISIBLE);
            holder.tv_status.setVisibility(View.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }

    public static class HotMovieViewHolder extends RecyclerView.ViewHolder {

        private SimpleDraweeView simpleDraweeView;
        private TextView tv_name, tv_year, tv_location, tv_type, tv_douban, tv_actor, tv_status;

        public HotMovieViewHolder(View itemView) {
            super(itemView);
            simpleDraweeView = (SimpleDraweeView) itemView.findViewById(R.id.simpleDraweeView);
            tv_name = (TextView) itemView.findViewById(R.id.tv_name);
            tv_year = (TextView) itemView.findViewById(R.id.tv_year);
            tv_location = (TextView) itemView.findViewById(R.id.tv_location);
            tv_type = (TextView) itemView.findViewById(R.id.tv_type);
            tv_douban = (TextView) itemView.findViewById(R.id.tv_douban);
            tv_actor = (TextView) itemView.findViewById(R.id.tv_actor);
            tv_status = (TextView) itemView.findViewById(R.id.tv_status);
        }
    }
}

File: app/src/main/java/cf/movie/slmovie/base/BaseMovies/model/BaseMoviesAdapter.kt
package cf.movie.slmovie.base.BaseMovies.model

import android.net.Uri
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import cf.movie.slmovie.R
import cf.movie.slmovie.base.BaseMovies.bean.BaseMoviesBean
import com.facebook.drawee.view.SimpleDraweeView
import java.util.*

/**
 * Created by 包俊 on 2017/7/21.
 */

class BaseMoviesAdapter(var movies: ArrayList<BaseMoviesBean.movie>) : RecyclerView.Adapter<BaseMoviesAdapter.HotMovieViewHolder>() {

    val serialVersionUID: Long = -2579386368620865598L

    fun refresh(movies: ArrayList<BaseMoviesBean.movie>) {
        this.movies = movies
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HotMovieViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_hotmovie, parent, false)
        return HotMovieViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: HotMovieViewHolder, position: Int) {
        val movie = movies!![position]
        val uri = Uri.parse(movie.post)
        holder.simpleDraweeView.setImageURI(uri)
        holder.tv_name.text = movie.name
        holder.tv_year.text = "上映年代：" + movie.year!!
        holder.tv_douban.text = "评分：" + movie.douban!!
        if (movie.details != null) {
            holder.tv_location.text = "地区：" + movie.details!!.location!!
            holder.tv_actor.text = "主演：" + movie.details!!.actor!!
            if (movie.details!!.isTV) {
                holder.tv_status.text = movie.details!!.status
                holder.tv_type.visibility = View.GONE
                holder.tv_status.visibility = View.VISIBLE
            } else {
                holder.tv_type.text = "类型：" + movie.details!!.type!!
                holder.tv_type.visibility = View.VISIBLE
                holder.tv_status.visibility = View.GONE
            }
            holder.tv_location.visibility = View.VISIBLE
            holder.tv_actor.visibility = View.VISIBLE
        } else {
            holder.tv_location.visibility = View.GONE
            holder.tv_actor.visibility = View.GONE
            holder.tv_type.visibility = View.GONE
            holder.tv_status.visibility = View.GONE
        }
    }

    override fun getItemCount(): Int {
        return movies!!.size
    }

    fun getMovies(position: Int): BaseMoviesBean.movie {
        return movies!![position]
    }

    class HotMovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var simpleDraweeView = itemView.findViewById(R.id.simpleDraweeView) as SimpleDraweeView
        var tv_name = itemView.findViewById(R.id.tv_name) as TextView
        var tv_year = itemView.findViewById(R.id.tv_year) as TextView
        var tv_location = itemView.findViewById(R.id.tv_location) as TextView
        var tv_type = itemView.findViewById(R.id.tv_type) as TextView
        var tv_douban = itemView.findViewById(R.id.tv_douban) as TextView
        var tv_actor = itemView.findViewById(R.id.tv_actor) as TextView
        var tv_status = itemView.findViewById(R.id.tv_status) as TextView
    }
}
